### PR TITLE
chore: move firestore to use GAPICBazel and regenerate

### PR DESCRIPTION
**Pull request by crwilcox**

### CODE REVIEW

File: synth.py [hunk 1]
```diff
@@ -32,18 +32,22 @@
...
+        bazel_target=f"//google/firestore/{version}:firestore-{version}-py",
     )
 
     s.move(
-        library / f"google/firestore_{version}",
+        library / f"google/cloud/firestore_{version}",
         f"google/cloud/firestore_{version}",
-        excludes=[ library / f"google/firestore_{version}/__init__.py"]
+        excludes=[
+            library / f"google/firestore_{version}/__init__.py"
```
Comment (tseaver) [inline synth.py:42]: Does this need to include `/cloud`?  E.g.:
```suggestion
            library / f"google/cloud/firestore_{version}/__init__.py"
```

File: synth.py [hunk 2]
```diff
@@ -52,62 +56,25 @@
...
-        f"google.cloud.firestore_admin_v1",
+        # proto_path=f"google/firestore/admin/{version}",
     )
+    s.move(
+        library / f"google/cloud/admin_{version}",
+        f"google/cloud/firestore_admin_{version}",
+         excludes=[
+            library / f"/google/cloud/admin_{version}/__init__.py"
+        ])
+    s.move(library / f"tests", f"tests")
```
Comment (tseaver) [inline synth.py:69]: Do we not need to move the `scripts`?

File: synth.py [hunk 3]
```diff
@@ -32,18 +32,22 @@
...
+            library / f"google/firestore_{version}/__init__.py"
+        ]
     )
-    
-    s.move(library / "scripts" )
 
+    s.move(
+        library / f"tests/",
+        f"tests",
+    )
```
Comment (tseaver) [inline synth.py:49]: `black` probably wants that on a single line.

File: google/cloud/firestore_admin_v1/py.typed [hunk 4]
```diff
@@ -1,2 +1,2 @@
 # Marker file for PEP 561.
-# The google-firestore-admin package uses inline types.
+# The google-cloud-firestore-admin package uses inline types.
```
Comment (crwilcox) [inline google/cloud/firestore_admin_v1/py.typed:2]: I am guessing this is not easily changed for multi-gen packages?

File: synth.py (hunk 2 above)
Comment (crwilcox) [reply synth.py:65]: Unsure of it, but I can bring them back in.

**Comment by crwilcox** [comment]
New Unit test failures that are inline with what @tseaver  was seeing.

```
=========================== short test summary info ============================
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_client_client_options[FirestoreClient-FirestoreGrpcTransport-grpc]
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_client_client_options[FirestoreAsyncClient-FirestoreGrpcAsyncIOTransport-grpc_asyncio]
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_client_client_options_scopes[FirestoreClient-FirestoreGrpcTransport-grpc]
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_client_client_options_scopes[FirestoreAsyncClient-FirestoreGrpcAsyncIOTransport-grpc_asyncio]
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_client_client_options_credentials_file[FirestoreClient-FirestoreGrpcTransport-grpc]
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_client_client_options_credentials_file[FirestoreAsyncClient-FirestoreGrpcAsyncIOTransport-grpc_asyncio]
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_client_client_options_from_dict
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_list_collection_ids
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_list_collection_ids_async
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_base_transport
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_base_transport_with_credentials_file
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_auth_adc
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_transport_auth_adc
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_grpc_transport_channel_mtls_with_client_cert_source
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_grpc_asyncio_transport_channel_mtls_with_client_cert_source
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_grpc_transport_channel_mtls_with_adc[mtls.squid.clam.whelk]
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_grpc_transport_channel_mtls_with_adc[mtls.squid.clam.whelk:443]
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_grpc_asyncio_transport_channel_mtls_with_adc[mtls.squid.clam.whelk]
FAILED tests/unit/gapic/firestore_v1/test_firestore_v1.py::test_firestore_grpc_asyncio_transport_channel_mtls_with_adc[mtls.squid.clam.whelk:443]
19 failed, 1100 passed, 6 warnings in 14.11s
nox > Command py.test --quiet --cov=google.cloud.firestore --cov=google.cloud --cov=tests.unit --cov-append --cov-config=.coveragerc --cov-report= --cov-fail-under=0 tests/unit failed with exit code 1
nox > Session unit-3.8 failed.

```

**Comment by crwilcox** [comment]
Coverage is down, but otherwise tests are good after merging the two test files and using the generated one as recommended by generator team

```
nox > Session docfx was successful.
nox > Ran multiple sessions:
nox > * lint: success
nox > * blacken: success
nox > * lint_setup_py: success
nox > * unit-3.6: success
nox > * unit-3.7: success
nox > * unit-3.8: success
nox > * system-3.7: success
nox > * cover: failed
nox > * docs: success
nox > * docfx: success
cleanup
```

**Comment by crwilcox** [comment]
I think coverage hasn't fell, so much as it was already too low. This issue is still open from @rafilong 's work. https://github.com/googleapis/python-firestore/issues/92

File: google/cloud/firestore_admin_v1/py.typed (hunk 4 above)
Comment (busunkim96) [reply google/cloud/firestore_admin_v1/py.typed:2]: I'm going to put in feature request to allow for custom package names. Sometimes I want to change the name for other reasons (to insert a hyphen in the middle of a reallylongapiname for example). Getting this right is important for `_GAPIC_LIBRARY_VERSION`.

**Review by busunkim96 — APPROVED** [review]
LGTM, please also bump the dep `google-api-core>=1.22.1` in setup.py.

**Comment by tseaver** [comment]
@crwilcox Do you want to force-merge this and then scramble to fix coverage?  Or add a temporary hack to `synth.py` to drop the required coverage level?

File: google/cloud/firestore_admin_v1/services/firestore_admin/async_client.py [hunk 5]
```diff
@@ -28,8 +28,8 @@
 from google.auth import credentials  # type: ignore
 from google.oauth2 import service_account  # type: ignore
 
-from google.api_core import operation as ga_operation  # type: ignore
-from google.api_core import operation_async  # type: ignore
+from google.api_core import operation as ga_operation
+from google.api_core import operation_async
```
Comment (tseaver) [inline google/cloud/firestore_admin_v1/services/firestore_admin/async_client.py:32]: This is going to break `pytype` -- I can't tell why it isn't being run in https://source.cloud.google.com/results/invocations/02606fdf-7059-497e-931f-b5c4192081d1/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fpython-firestore%2Fpresubmit%2Fpresubmit/log

The fix for it is in [this PR], released in [`gapic-generator-python 0.33.2`](https://github.com/googleapis/gapic-generator-python/releases/tag/v0.33.2).  Do we need to update the bazel config to use the newer version?

File: google/cloud/firestore_admin_v1/services/firestore_admin/client.py [hunk 6]
```diff
@@ -27,12 +28,13 @@
...
+from google.auth.transport.grpc import SslCredentials  # type: ignore
 from google.auth.exceptions import MutualTLSChannelError  # type: ignore
 from google.oauth2 import service_account  # type: ignore
 
-from google.api_core import operation as ga_operation  # type: ignore
-from google.api_core import operation  # type: ignore
-from google.api_core import operation_async  # type: ignore
+from google.api_core import operation as ga_operation
+from google.api_core import operation
+from google.api_core import operation_async
```
Comment (tseaver) [inline google/cloud/firestore_admin_v1/services/firestore_admin/client.py:37]: Likewise here.

**Comment by crwilcox** [comment]
@tseaver I forced it in. We have an issue to fix coverage that should likely be prioritized. Though I think if it blocks anything else I am okay dropping to 97% for a time

File: scripts/fixup_firestore_v1_keywords.py [hunk 7]
```diff
@@ -55,6 +55,7 @@ class firestoreCallTransformer(cst.CSTTransformer):
     'run_query': ('parent', 'structured_query', 'transaction', 'new_transaction', 'read_time', ),
     'update_document': ('document', 'update_mask', 'mask', 'current_document', ),
     'write': ('database', 'stream_id', 'writes', 'stream_token', 'labels', ),
+
```
Comment (tseaver) [inline scripts/fixup_firestore_v1_keywords.py:58]: Odd that `black` didn't correct this -- do we not run it against the `scripts` directory?

File: setup.py [hunk 8]
```diff
@@ -25,7 +25,7 @@
 version = "2.0.0-dev1"
 release_status = "Development Status :: 5 - Production/Stable"
 dependencies = [
-    "google-api-core[grpc] >= 1.21.0, < 2.0.0dev",
+    "google-api-core[grpc] >= 1.22.1, < 2.0.0dev",
```
Comment (tseaver) [inline setup.py:28]: The `1.22.1` release is odd to have as a minimum:  it changes only docstrings.  Seems like we would want `1.22.0` (to pick up the feature, "allow quota project to be passed to create_channel"), or else `1.22.2` (bugfix, "only add quota project id if supported").

**Pull request merged**